### ISSUE TITLE
Reorder Default Export Columns

### ISSUE THREAD

**Issue by RFSH** [informatics-isi-edu/ermrestjs#786]
The default export template is using the order that columns are defined in the database. This is confusing since most of the times the `RID` column is pushed to the end of the list, and also a lot of unnecessary columns will be included. We should

- [x] ~~Add RID to the beginning of the projection list~~ (We're already adding the shortestkey first. Unless the table doesn't have any foreignkeys which in that case we will fall back to using the entity `api`).
- [x] Use the `visible-columns` in `export` context (if it's not defined we should fall back to `detailed`) for getting the list of columns. This is how we should treat each visible column type:
  - Normal Columns: add to the projection list.
  - ForeignKey pseudo-column: Add the constituent columns alongside extra candidate columns (if needed).
  - Key pseudo-column: add the constituent columns.
  - Asset pseudo-column: add all the metadata columns alongside the url value.
  - Inline table: not applicable. Because of the one-to-many nature of the relationship this is not feasible. 
  - other pseudo-columns (aggregate): not applicable.

**Comment by RFSH** [comment]
This has been merged with master. Assigning to @animesh-singh for verification.

With the recent change we modified the default export heuristics to use the `export` context in `visible-columns` annotation instead of the table columns. If the context is not defined, it will fallback to the `detailed` annotation. 

Link to [the export document](https://github.com/informatics-isi-edu/ermrestjs/blob/master/docs/user-docs/export.md).
